test/Microsoft.AspNetCore.Server.KestrelTests/ChunkedRequestTests.cs

```diff
@@ -0,0 +1,294 @@
+                }
+
+                requestsReceived++;
+
+                response.Headers.Clear();
+                response.Headers["Content-Length"] = new[] { "11" };
+
+                await response.Body.WriteAsync(Encoding.ASCII.GetBytes("Hello World"), 0, 11);
+            }, testContext))
+            {
+                var response = string.Join("\r\n", new string[] {
+                    "HTTP/1.1 200 OK",
+                    "Content-Length: 11",
+                    "",
+                    "Hello World"});
+
+                var expectedFullResponse = string.Join("", Enumerable.Repeat(response, requestCount + 1));
+
+                using (var connection = new TestConnection(server.Port))
+                {
+                    await connection.Send(
+                        "POST / HTTP/1.1",
+                        "Transfer-Encoding: chunked",
+                        "",
+                        "C", "HelloChunked",
+                        "0",
+                        "");
+
+                    for (var i = 1; i < requestCount; i++)
+                    {
+                        await connection.Send(
+                            "POST / HTTP/1.1",
+                            "Transfer-Encoding: chunked",
+                            "",
+                            "C", "HelloChunked",
+                            "0",
+                            string.Concat("X-Trailer-Header", new string('a', i)),
+                            "");
+                    }
+
+                    await connection.SendEnd(
+                        "POST / HTTP/1.1",
+                        "Content-Length: 7",
+                        "",
+                        "Goodbye");
+
+                    await connection.ReceiveEnd(expectedFullResponse);
+                }
+            }
+        }
+
+        private class TestApplicationErrorLogger : ILogger
+        {
+            public int ApplicationErrorsLogged { get; set; }
+
+            public IDisposable BeginScopeImpl(object state)
+            {
+                return new Disposable(() => { });
+            }
+
+            public bool IsEnabled(LogLevel logLevel)
+            {
+                return true;
+            }
+
+            public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Exception exception, Func<TState, Exception, string> formatter)
+            {
+                // Application errors are logged using 13 as the eventId.
+                if (eventId.Id == 13)
+                {
+                    ApplicationErrorsLogged++;
+                }
+            }
+        }
+
+        private class PassThroughConnectionFilter : IConnectionFilter
+        {
+            public Task OnConnectionAsync(ConnectionFilterContext context)
+            {
+                context.Connection = new LoggingStream(context.Connection, new TestApplicationErrorLogger());
+                return TaskUtilities.CompletedTask;
+            }
+        }
+    }
+}
+
```

Comment:
Having issues with feeds and running tests (to test the tests) https://github.com/aspnet/Home/issues/1289